src/date.c

```diff
@@ -363,22 +365,7 @@ static int d_parsedatestr(const char *jstr, size_t len, struct timeparts *tp,
     }
   }
 
-  /* #2: trying getdate() - it never works on some OS, but... */
-  {
-    struct tm *tm = getdate(str);
-    if (tm != NULL) {
-      tp->year = tm->tm_year + 1900;
-      tp->month = tm->tm_mon;
-      tp->day = tm->tm_mday;
-      tp->hour = tm->tm_hour;
-      tp->min = tm->tm_min;
-      tp->sec = tm->tm_sec;
-
-      return 1;
-    }
-  }
-
-  /* #3: trying toString()/toUTCString()/toDateFormat() formats */
+  /* trying toString()/toUTCString()/toDateFormat() formats */
   {
     char month[4];
     const char *frmString = " %03*s %03s %02d %d %02d:%02d:%02d %03s%d";
```

Comment:
I tried with `Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)`
and `clang version 3.5.0 (tags/RELEASE_350/final)`

Anyway, I get the warnings from the compiler but the actual libsystem sscanf implementation does handle it.

However I'm not sure it works as intended anyway. Namely, when you use `%03*s` it will consume 3 chars and then leave whatever is left to be consumed as the next token, without affecting the actual sscanf return value, so you cannot use this in order to ensure that a three letter DOW token is present.

For example, by using `%03*s` it will actually parse `IncDec 01 2015 00:00:00` as `Tue Dec ...`,
while by having `%*s` it correctly fail to parse that, but it will accept `Foooo Dec 01 2015 00:00:00`.
On the other hand, the current `%03*s` format also does accept DOWs shorter than 3 letters, which is suboptimal as well.

Probably a better way of handling this is:

```c
    const char *frmString = " %*s%n %03s %02d %d %02d:%02d:%02d %03s%d";
    res = sscanf(str, frmString, &dowlen, month, &tp->day, &tp->year,
                 &tp->hour, &tp->min, &tp->sec, gmt, tz);
```

and check that `dowlen == 3`.

Anyway I don't think this is very important, I'm more concerned about becoming warning blind.

Here the compiler doesn't warn via `-Wformat-invalid-specifier`, because the format string is not a literal parameter of `sscanf`, but ASAN issues a runtime warning.